File: Easy/2558-take-gifts-from-the-richest-pile/try-3.py
/*
 * Problem #2558: Take Gifts From the Richest Pile
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: python3
 * Date: 12/12/2024, 22:47:09
 * Link: https://leetcode.com/problems/take-gifts-from-the-richest-pile/
 */

class Solution:
    def pickGifts(self, gifts: List[int], k: int) -> int:
        num = [-x for x in gifts]
        heapify(num)
        while k :
            t = int(sqrt(-heappop(num)))
            heappush(num,-t)
            k-=1
        return -sum(num)

File: Easy/191-number-of-1-bits/try-3.cpp
/*
 * Problem #191: Number of 1 Bits
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: cpp
 * Date: 27/10/2025, 15:14:46
 * Link: https://leetcode.com/problems/number-of-1-bits/
 */

class Solution {
public:
    int hammingWeight(int n) {
        int res = 0;
        while(n){
            n = n & (n-1);
            res++;
        }
        return res;
    }
};
